asset_dashboard/models.py

```diff
@@ -64,33 +64,42 @@ def __str__(self):
 class ProjectScore(models.Model):
 
     project = models.OneToOneField(Project, on_delete=models.CASCADE)
-    core_mission_score = models.IntegerField(default=1,
+    core_mission_score = models.IntegerField(null=True, blank=True,
                                              validators=[MinValueValidator(1),
                                                          MaxValueValidator(5)])
-    operations_impact_score = models.IntegerField(default=1,
+    operations_impact_score = models.IntegerField(null=True, blank=True,
                                                   validators=[MinValueValidator(1),
                                                               MaxValueValidator(5)])
-    sustainability_score = models.IntegerField(default=1,
+    sustainability_score = models.IntegerField(null=True, blank=True,
                                                validators=[MinValueValidator(1),
                                                            MaxValueValidator(5)])
-    ease_score = models.IntegerField(default=1,
+    ease_score = models.IntegerField(null=True, blank=True,
                                      validators=[MinValueValidator(1),
                                                  MaxValueValidator(5)])
-    geographic_distance_score = models.IntegerField(default=1,
+    geographic_distance_score = models.IntegerField(null=True, blank=True,
                                                     validators=[MinValueValidator(1),
                                                                 MaxValueValidator(5)])
-    social_equity_score = models.IntegerField(default=1,
+    social_equity_score = models.IntegerField(null=True, blank=True,
                                               validators=[MinValueValidator(1),
                                                           MaxValueValidator(5)])
 
     @property
     def total_score(self):
-        '''return the total, weighted score'''
-        ...
+        # get the ProjectScore fields, excluding the id and project fields
+        score_fields = [f for f in self._meta.get_fields() if f.name != 'id' if f.name != 'project']
+        score_weights = ScoreWeights.objects.all()[0]
+        total_score = 0
+
+        for field in score_fields:
+            score_field_value = field.value_from_object(self) if field.value_from_object(self) else 0
```

Comment:
if a single field element is undefined, then the *total score* should be 0

Reply:
ah, yeah. that makes sense.

Reply:
fixed here: d11687a